applications/FluidDynamicsApplication/custom_strategies/strategies/residualbased_predictorcorrector_velocity_bossak_scheme_turbulent.h

```diff
@@ -398,14 +419,24 @@ namespace Kratos {
                         array_1d<double, 3 > & OldDisplacement = (itNode)->FastGetSolutionStepValue(DISPLACEMENT, 1);
                         array_1d<double, 3 > & CurrentDisplacement = (itNode)->FastGetSolutionStepValue(DISPLACEMENT, 0);
 
-                        noalias(itNode->FastGetSolutionStepValue(MESH_VELOCITY)) = itNode->FastGetSolutionStepValue(VELOCITY);
-                        UpdateDisplacement(CurrentDisplacement, OldDisplacement, OldVelocity, OldAcceleration, CurrentAcceleration);
+                  if((itNode)->FastGetSolutionStepValue(IS_LAGRANGIAN_INLET) == 0.0)
+			{
+			    noalias(itNode->FastGetSolutionStepValue(MESH_VELOCITY)) = itNode->FastGetSolutionStepValue(VELOCITY);
+			    UpdateDisplacement(CurrentDisplacement, OldDisplacement, OldVelocity, OldAcceleration, CurrentAcceleration);
+			}
+			else
+			{
+			  itNode->FastGetSolutionStepValue(MESH_VELOCITY_X) = 0.0;
+			  itNode->FastGetSolutionStepValue(MESH_VELOCITY_Y) = 0.0;
+			  itNode->FastGetSolutionStepValue(DISPLACEMENT_X) = 0.0;
+			  itNode->FastGetSolutionStepValue(DISPLACEMENT_Y) = 0.0;
+			}
                     }
                 }
             }
 
-            // if (rModelPart.GetCommunicator().MyPID() == 0)
-            //     std::cout << "end of prediction" << std::endl;
+             if (rModelPart.GetCommunicator().MyPID() == 0)
+                 std::cout << "end of prediction" << std::endl;
```

Comment:
These kinds of prints are fine for debug, but we would prefer not to have them in the final version. Please comment them out again or replace them by calls to the kratos logger.

Reply:
@jcotela 

done dear, these two lines are commented now,